tests/test_kwargs_and_defaults.py

```diff
@@ -23,6 +23,34 @@ def test_function_signatures(doc):
         doc(m.KWClass.foo1)
         == "foo1(self: m.kwargs_and_defaults.KWClass, x: int, y: float) -> None"
     )
+    assert (
+        doc(m.kw_lb_func0)
+        == "kw_lb_func0(custom: m.kwargs_and_defaults.CustomRepr = Hello World!) -> None"
```

Comment:
It's not "real" that a string repr comes out without quotes.
It doesn't really matter, but to not have something that looks too artificial (and gets people scratching their head until they figure it out), I'd try to emulate a real-world situation more closely, within reason.
In this case, I'd try to simply not use something that looks like a string.
Maybe: take an actual Eigen repr, simplify/shorten, then add in whitespace like you do here.

But there is also a more substantial question: **What if there is a str repr with newlines?** We don't want to change that.

Reply:
You're right, this looks confusingly like a string.

I'm not sure I understand the second part correctly. These changes only effect the repr within the automatically generated docstring signatures. The repr of objects outside of this context stay the same. If your concern is the removal of intentional newlines within a string default value in the human-readable documentation, I think the broken generated stubs and/or Sphinx documentation would be more confusing to users. Personally, it took me quite a while to figure out what was going wrong.

Alternatively, it might be possible to check the `argument_record` for string types and not apply the stripping function at the cost of incompatibility with the stub and documentation tools for this specific case.

Reply:
> If your concern is the removal of intentional newlines within a string default value in the human-readable documentation,

Yeah, that's where my mind was going.

> I think the broken generated stubs and/or Sphinx documentation would be more confusing to users.

I agree.

> Alternatively, it might be possible to check the `argument_record` for string types and not apply the stripping function

That sounds like opening a can of worms.

Quoting again:
> intentional newlines within a string default value

Could you please add a test case? — I'm assuming that's really easy and then we have something concrete to look at. And others will see that we thought about it.

IIUC with the current state of the PR the newline will get replaced with a space.

Wondering:

* How about inserting something like `<nl>` instead?

* Should we do that unconditionally (also for the Eigen case)?

* Or something like: if we see any quote (`'` or `"`) in the docstring, we use `<nl>`, otherwise we use a space?

(I don't want to turn this into a science project, but also not sweep this under the rug without thinking a least a little bit.)

Reply:
I like the idea of adding something like `<nl>`. It might look a little confusing for the Eigen case (see [here](https://jeanelsner.github.io/panda-py/panda_py.controllers.html#panda_py.controllers.CartesianImpedance) for a real-life documentation example).

Your last suggestion seems like a nice compromise to me. How would propose the "parsing" would look like? Simply checking for the presence of quotes, checking the first and last characters or something more sophisticated even?

Reply:
What I had in mind was simply `s.find('\'') || s.find('"')`

Since you like the `<nl>` idea: I'd give it ~5 minutes looking around for prior art (`<NL>`, `<newline>`, ...) and emulate that. But not more than a few minutes.

Reply:
I added a test case and it turns out string representations don't include the newlines anyway, but rather print `\n` literally.